src/metax_api/models/catalog_record.py

```diff
@@ -888,6 +900,9 @@ def _pre_update_operations(self):
                 self._handle_preferred_identifier_changed()
 
     def _post_update_operations(self):
+        if get_identifier_type(self.research_dataset['preferred_identifier']) == IdentifierType.DOI:
```

Comment:
To be precise, datacite update only needs to be done when contents of research_dataset changes? When changing PAS fields for example, they probably wouldn't care that DOI service was not responding. Probably most convenient to set a flag for this block to execute in pre_update_operations, when research_dataset content change is checked.